### PR TITLE
Shifts model

### PR DESCRIPTION
**Pull request by jansm04**
Implemented GET endpoints `getShiftsByVolunteerId` and `getShiftsByDate`.

Resolved merge conflicts with `class-model` (#7) by converting `shiftController.ts` from a class to a module, and merging all functions in conflicting files.

### CODE REVIEW

File: backend/src/routes/shiftRoutes.ts [hunk 1]
```diff
@@ -1,13 +1,26 @@
-import express from 'express';
-import { Request, Response } from 'express';
-import ShiftController from '../controllers/shiftController.js';
+import { Router, Request, Response } from 'express';
```
Comment (jjessieshang) [inline backend/src/routes/shiftRoutes.ts:2]: remove empty line here

File: backend/src/controllers/shiftController.ts [hunk 2]
```diff
@@ -1,21 +1,89 @@
...
+
+    try {
+        const shifts = await shiftModel.getShiftsByDate(shift_date);
+        res.status(200).json(shifts);
+    } catch (error) {
+        return res.status(500).json({
+            error: `Internal server error. ${error}`
+        });
+    }
+
```
Comment (jjessieshang) [inline backend/src/controllers/shiftController.ts:82]: remove empty line

**Review by jjessieshang — APPROVED** [review]
Tiny formatting thing. Otherwise both routes look good to me! Thanks for restructuring the controller.

File: backend/src/controllers/shiftController.ts [hunk 3]
```diff
@@ -1,21 +1,89 @@
...
+            error: "Missing required parameter: 'volunteer_id'"
+        });
+    }
+
+    try {
+        const shifts = await shiftModel.getShiftsByVolunteerId(volunteer_id);
+        res.status(200).json(shifts);
+    } catch (error) {
+        return res.status(500).json({
+            error: `Internal server error. ${error}`
```
Comment (khaidoan2729) [inline backend/src/controllers/shiftController.ts:52]: @jjessieshang told me to use ${error.message} to display the message

File: backend/src/controllers/shiftController.ts [hunk 4]
```diff
@@ -1,21 +1,89 @@
...
+            error: "'shiftDate' must be a valid date of the format 'YYYY-MM-DD'"
+        });
+    }
+
+    try {
+        const shifts = await shiftModel.getShiftsByDate(shift_date);
+        res.status(200).json(shifts);
+    } catch (error) {
+        return res.status(500).json({
+            error: `Internal server error. ${error}`
```
Comment (khaidoan2729) [inline backend/src/controllers/shiftController.ts:79]: same here.

**Review by khaidoan2729 — APPROVED** [review]
Looks good other than error handling.

**Comment by jjessieshang** [comment]
Can we try using interfaces? For the requests which pass parameters in the body, even if it's just a couple of fields it'll help with consistency in the code. 

Example of using an interface: https://github.com/ubclaunchpad/neuron/blob/26bb7c084c5e94b3b400947ed0492f30b9fc8bad/backend/src/controllers/instructorController.ts#L37

^Not sure if we can cast the req.body directly to an interface if not all fields are present, but let's see.

Don't need to create the interface, there's a PR for this in progress: https://github.com/ubclaunchpad/neuron/pull/18

**Pull request merged**